pkg/webhooks/clustertrainingruntime_webhook.go

```diff
@@ -48,7 +51,15 @@ func (w *ClusterTrainingRuntimeWebhook) ValidateCreate(ctx context.Context, obj
 	clTrainingRuntime := obj.(*trainer.ClusterTrainingRuntime)
 	log := ctrl.LoggerFrom(ctx).WithName("clustertrainingruntime-webhook")
 	log.V(5).Info("Validating create", "clusterTrainingRuntime", klog.KObj(clTrainingRuntime))
-	return nil, validateReplicatedJobs(clTrainingRuntime.Spec.Template.Spec.ReplicatedJobs).ToAggregate()
+	var warnings admission.Warnings
+	if labelutil.IsSupportDeprecated(clTrainingRuntime.Labels) {
+		warnings = append(warnings, fmt.Sprintf(
+			"ClusterTrainingRuntime \"%s\" is deprecated and will be removed in a future release of Kubeflow Trainer. See runtime deprecation policy: %s",
+			clTrainingRuntime.Name,
+			constants.RuntimeDeprecationPolicyURL,
+		))
+	}
+	return warnings, validateReplicatedJobs(clTrainingRuntime.Spec.Template.Spec.ReplicatedJobs).ToAggregate()
```

Comment:
This is not necessary. Warnings should only be emitted for TrainJobs referencing deprecated runtimes, not for the creation of ClusterTrainingRuntimes themselves.

Reply:
@andreyvelich ah you're right, I missed this. That makes sense.